service/resource/vaultpki/delete.go

```diff
@@ -0,0 +1,66 @@
+package vaultpki
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/framework/context/deletionallowedcontext"
+
+	"github.com/giantswarm/cert-operator/service/key"
+)
+
+func (r *Resource) GetDeleteState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	currentVaultPKIState, err := toVaultPKIState(currentState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	desiredVaultPKIState, err := toVaultPKIState(desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var vaultPKIStateToDelete VaultPKIState
+	if deletionallowedcontext.IsDeletionAllowed(ctx) {
```

Comment:
Note that we do not allow deletion of PKI backends by the operator yet because this is still done by the cluster-service. Reason for this is our insufficient TPR. We will iterate on this in the near future. For now deletion is not allowed because nobody is setting the state to the context and the default is false.